### PR TITLE
Coalesced background feed loading

### PR DESCRIPTION
**Pull request by joewalsh**
Feed updates are done on a background managed object context and the UI is updated in a single pass when those changes are merged into the view. `WMFArticleDataStore` and `WMFContentGroupDataStore` were removed and replaced with categories on `NSManagedObjectContext` and the associated object classes. The feed also now automatically loads older content (if less than 30 days old) when the user scrolls past 80% of the existing content.

https://phabricator.wikimedia.org/T162167
https://phabricator.wikimedia.org/T142577

### CODE REVIEW

**Comment by montehurd** [comment]
Seeing a good deal more red than green on the diff... I like it.

File: Wikipedia/Code/WMFExploreViewController.m [hunk 1]
```diff
@@ -682,6 +686,14 @@ - (WMFLayoutEstimate)collectionView:(UICollectionView *)collectionView estimated
         } break;
         case WMFFeedDisplayTypePageWithPreview: {
             WMFArticle *article = [self articleForIndexPath:indexPath];
+            NSString *key = article.key;
+            NSString *cacheKey = [NSString stringWithFormat:@"%@-%lli", key, (long long)columnWidth];
+            NSNumber *cachedValue = [self.cachedHeights objectForKey:cacheKey];
+            if (cachedValue) {
+                estimate.height = [cachedValue doubleValue];
```
Comment (montehurd) [inline Wikipedia/Code/WMFExploreViewController.m]: Will this play nice with rotation and Dynamic Type changes?

Comment (joewalsh) [inline Wikipedia/Code/WMFExploreViewController.m]: @montehurd  good catch, I addressed this in the separate PR I opened for this change

**Comment by coreyfloyd** [comment]
Just wanted to say this is awesome… so happy this is going in! 💪 💯

**Pull request merged**